services/core-api/tests/mines/tailings/resources/test_mine_tailings_resource.py

```diff
@@ -40,40 +39,89 @@ def test_post_first_mine_tailings_storage_facility_by_mine_guid(test_client, db_
                                                                 auth_headers):
     mine = MineFactory(minimal=True)
     data = {
-        'mine_tailings_storage_facility_name':'a name',
+        'mine_tailings_storage_facility_name': 'a name',
         'latitude': '50.6598000',
-        'longitude': '-120.5134000',  
+        'longitude': '-120.5134000',
         'consequence_classification_status_code': 'LOW',
         'tsf_operating_status_code': 'OPT',
         'has_itrb': True,
     }
     assert len(mine.mine_tailings_storage_facilities) == 0
 
     post_resp = test_client.post(
-        f'/mines/{mine.mine_guid}/tailings',
-        data=data,
-        headers=auth_headers['full_auth_header'])
+        f'/mines/{mine.mine_guid}/tailings', data=data, headers=auth_headers['full_auth_header'])
     assert post_resp.status_code == 201
     assert len(mine.mine_tailings_storage_facilities) == 1
 
 
 def test_post_first_mine_tailings_storage_facility_by_mine_guid_creates_tsf_required_reports(
-    test_client, db_session, auth_headers):
+        test_client, db_session, auth_headers):
     mine = MineFactory(minimal=True)
     data = {
-        'mine_tailings_storage_facility_name':'a name',
+        'mine_tailings_storage_facility_name': 'a name',
         'latitude': '50.6598000',
-        'longitude': '-120.5134000',  
+        'longitude': '-120.5134000',
         'consequence_classification_status_code': 'LOW',
         'tsf_operating_status_code': 'OPT',
         'has_itrb': True,
     }
     assert len(mine.mine_tailings_storage_facilities) == 0
 
     post_resp = test_client.post(
-        f'/mines/{mine.mine_guid}/tailings',
-        data=data,
-        headers=auth_headers['full_auth_header'])
+        f'/mines/{mine.mine_guid}/tailings', data=data, headers=auth_headers['full_auth_header'])
     assert post_resp.status_code == 201
     tsf_required_reports = MineReportDefinition.find_required_reports_by_category('TSF')
     assert len(mine.mine_reports) == len(tsf_required_reports)
+
+
+def test_put_tailings_storage_facility_not_found(test_client, db_session, auth_headers):
+    tsf_updated = MineTailingsStorageFacilityFactory()
+
+    data = {
+        'mine_tailings_storage_facility_name': tsf_updated.mine_tailings_storage_facility_name,
+        'latitude': tsf_updated.latitude,
+        'longitude': tsf_updated.longitude,
+        'consequence_classification_status_code':
+        tsf_updated.consequence_classification_status_code,
+        'tsf_operating_status_code': tsf_updated.tsf_operating_status_code,
+        'has_itrb': tsf_updated.has_itrb,
+    }
+
+    put_resp = test_client.put(
+        f'/mines/{uuid.uuid4()}/tailings/{uuid.uuid4()}',
+        data=data,
+        headers=auth_headers['full_auth_header'])
+    put_data = json.loads(put_resp.data.decode())
+
+    assert put_resp.status_code == 404
+    assert 'not found' in put_data['message']
```

Comment:
No need to do this assertion, we do not do it in any other tests. It is ok just to check the status code